### DISCUSSION TITLE
noindex on specific page

### DISCUSSION THREAD

**Discussion by jcalabreseSH** [squidfunk/mkdocs-material#4184] [Q&A]
I have a hidden page where I would like to add in the header     
`<meta name="robots" content="noindex,noarchive,nofollow" />`

The goal is to not have any meta for crawlers or google reading this page since we want it hidden. This page currently showing on the site map.
Is there a way to add this into the header on specific pages?

**Comment by squidfunk** [comment]
Just add the tag directly inside the Markdown page. `meta` tags don't need to be part of the `head`.

**Reply by MaximilianKohler** [reply]
So it supports this by default?

```
---
title: This is my homepage
description: I worked hard on this site
hide:
  - navigation
robots: noindex,noarchive,nofollow
---
```
Is that the correct format?

I saw some more complicated directions in other discussions

**Reply by squidfunk** [reply]
No. That is not supported. But you can just use:

``` md
---
title: This is my homepage
description: I worked hard on this site
hide:
  - navigation
---

<meta name="robots" content="noindex, noarchive, nofollow">

# Document title
```

**Reply by kamilkrzyskow** [reply]
Even though meta tags are valid in the body of a page, I would personally not put a `robots` tag outside of the `<head>` tag. 

\[EDIT: Found that the docs have an example with robots https://squidfunk.github.io/mkdocs-material/reference/#customization \]

In the [mkdocs-nype](https://github.com/nypesap/mkdocs-nype) theme I've added an option to add robots data via the page meta:
- https://github.com/nypesap/mkdocs-nype/blob/4d1dcef48b8c50b7ca5bf32e8c1583dd22cd62a8/mkdocs_nype/templates/nype-base.html#L32-L37

Or another addition allows to input custom tags in the `<head>` section, like `<script src="https://external"></script>`:
- https://github.com/nypesap/mkdocs-nype/blob/4d1dcef48b8c50b7ca5bf32e8c1583dd22cd62a8/mkdocs_nype/templates/nype-base.html#L64-L78

Those changes can be ported over via `main.html` override and `site_meta` or `extrahead` blocks:
- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
- https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks

**Reply by squidfunk** [reply]
Any reason why you shouldn't put `robots` outside of `head`?

**Reply by kamilkrzyskow** [reply]
Given there is the HTML5 standard, which allows to put `meta` tags anywhere in the file and they should be properly read, and given we are in the age of AI web scrapers, which load the whole page including JavaScript execution one could think that anything goes, right?

Well, I still believe there is value in putting hidden metadata into the header for web scrapers to stream text of the HTML file up to the end of the `</head>` section to decide faster what to do with the file, and after detecting a `noindex` ending the connection preventing unnecessary server load.

Perhaps this is a bad example which is unrelated, but I recalled an issue with [LinkedIn](https://github.com/squidfunk/mkdocs-material/issues/7294), where the Social card images stopped working in the preview. I remember that when investigating some of them worked, while others did not. This could've been a back-end issue or a bad HTML parser, which didn't read the source properly, and made some "shortcut".

Another such "shortcut" could be reading up to the `</head>` tag, so better to be on the safe side 😄 

>Any reason why you shouldn't put robots outside of head?

To answer the question, I'm a bad coder, which does a lot of hacks, so I can imagine making the mistake of not loading all of the `meta` tags on the page. Bad coder protection haha

**Reply by squidfunk** [reply]
Hmm, yeah, so I'm on the fence here between adhering to standards vs. writing code that is as defensive as possible to compensate for the missing knowledge of other developers. I'd say for the `noindex` case, `yourdomain.com/robots.txt` is an even better choice, but that might not work with GitHub Pages, not sure. Regardless, what's asked here can be achieved with overrides, or with a hook and maybe some metadata 😊

__Edit__: To be clear, "missing knowledge of other developers" was not meant to confirm that you're a bad coder. It was meant as a general note that not everybody is familiar with the latest web standards, which might also be too much of an ask. You are, in fact, an amazing coder with deep knowledge of MkDocs internals, and for sure the king of monkey-patching 🤟
